addons/calendar/static/src/views/calendar_form/calendar_quick_create.js

```diff
@@ -70,4 +70,20 @@ export class CalendarQuickCreate extends FormViewDialog {
             }
         });
     }
+
+    async onExpand() {
+        const beforeLeaveCallbacks = this.viewProps.__beforeLeave__.callbacks;
+        const res = await Promise.all(beforeLeaveCallbacks.map((callback) => callback()));
+        if (!res.includes(false)) {
+            this.actionService.doAction({
+                type: "ir.actions.act_window",
+                res_model: this.props.resModel,
+                res_id: this.currentResId,
+                views: [[false, "form"]],
+            },
+            {
+                additionalContext: this.props.context
+            });
+        }
+    }
```

Comment:
This is a copy paste from the form_view_dialog.js. strange that this part is not called. do you know why ?
If it's called without the proper context, can't you make it more clean by allowing to get the context somewhere else instead of copy pasting the whole method without calling super ? With that change, if async onExpand() is updaed in web, you will never get the bugfix here :-(